runtime/include/QuantumDevice.hpp

```diff
@@ -55,310 +57,558 @@ struct QuantumDevice {
+     * -> initialization on the same set of qubits.
+     * See `ReleaseQubit` for caveats around releasing / resetting entangled qubits.
+     *
+     * @param n Bitstring representation of the basis state |n>, stored as a Byte-array.
+     * @param wires The qubits to initialize.
      */
-    [[nodiscard]] virtual auto Zero() const -> Result = 0;
+    virtual void SetBasisState([[maybe_unused]] DataView<int8_t, 1> &n,
+                               [[maybe_unused]] std::vector<QubitIdType> &wires)
+    {
+        RT_FAIL("SetBasisState is unsupported by device");
+    }
 
     /**
-     * @brief Result value for "One"  used in the measurement process.
+     * @brief (Optional) Initialize qubits to an arbitrary quantum state.
      *
-     * @return `Result`
+     * Like `SetBasisState`, but instead of initializing to a single computational basis state
+     * this instruction works with a vector of complex amplitudes, one for each possible basis
+     * state.
+     *
+     * @param state Quantum state vector of size 2^len(wires).
+     * @param wires The qubits to initialize.
      */
-    [[nodiscard]] virtual auto One() const -> Result = 0;
+    virtual void SetState([[maybe_unused]] DataView<std::complex<double>, 1> &state,
+                          [[maybe_unused]] std::vector<QubitIdType> &wires)
+    {
+        RT_FAIL("SetState is unsupported by device");
+    }
+
+    // ----------------------------------------
+    //  QUANTUM OBSERVABLES
+    // ----------------------------------------
 
     /**
-     * @brief A helper method to print the state vector of a device.
+     * @brief (Optional) Construct a named observable.
+     *
+     * This operation instructs the device to generate a named observable based on the provided enum
+     * `id`, which will be one of {Identity, PauliX, PauliY, PauliZ, Hermitian}. The Hermitian kind
+     * uses additional data to define the observable from a 2D complex matrix, which is expected to
+     * be Hermitian. If this is not the case, the device is free to raise an error or produce
+     * undefined behaviour. Additionally, the operation accepts a list of target qubits whose length
+     * is expected to match the dimensionality of the observable.
+     *
+     * The observable system in the Catalyst Runtime relies primarily on the device's implementation
+     * for support. As such, the device is free to represent observables in any way they wish.
+     * Before using an observable in measurement processes, Catalyst will invoke one or more
+     * observable methods from the interface. The device is expected to process and cache the
+     * information in such a way that it is ready to use in a subsequent measurement process call.
+     * The device must return an ID than unambiguously identifies the requested observable, but this
+     * ID is not required to be unique across different calls supplied with the same information.
+     *
+     * @param id The name of the observable.
```

Comment:
```suggestion
     * @param id The type of the observable.
```

Reply:
I don't want to use type, it's a very overloaded word.